src/test/java/com/exasol/adapter/document/mapping/auto/SchemaInferencerTest.java

```diff
@@ -0,0 +1,89 @@
+
+@ExtendWith(MockitoExtension.class)
+class SchemaInferencerTest {
+
+    private static final String ADDITIONAL_CONFIG = "additionalConfig";
+    private static final String DESCRIPTION = "descr";
+    private static final String DESTINATION = "dest";
+    private static final String SOURCE = "mySource";
+    @Mock
+    SchemaFetcher schemaFetcherMock;
+    private SchemaInferencer inferencer;
+
+    @BeforeEach
+    void setUp() {
+        inferencer = new SchemaInferencer(schemaFetcherMock);
+    }
+
+    @Test
+    void mappingPresent() {
+        final EdmlDefinition definition = createDefinition(createMapping());
+        final EdmlDefinition result = inferencer.inferSchema(definition);
+        assertThat(result, sameInstance(definition));
+    }
+
+    @Test
+    void mappingNotPresent() {
+        final EdmlDefinition definition = createDefinition(null);
+        final MappingDefinition mapping = createMapping();
+        when(schemaFetcherMock.fetchSchema(SOURCE)).thenReturn(Optional.of(mapping));
+        final EdmlDefinition result = inferencer.inferSchema(definition);
+        assertAll(() -> assertThat(result, not(sameInstance(definition))),
+                () -> assertThat(result.getMapping(), sameInstance(mapping)),
+                () -> assertThat(result.getAdditionalConfiguration(), equalTo(ADDITIONAL_CONFIG)),
+                () -> assertThat(result.getSource(), equalTo(SOURCE)),
+                () -> assertThat(result.getDestinationTable(), equalTo(DESTINATION)),
+                () -> assertThat(result.getDescription(), equalTo(DESCRIPTION)),
+                () -> assertThat(result.isAddSourceReferenceColumn(), is(true)));
+    }
+
+    @Test
+    void mappingNotPresentSourceNotSupported() {
+        final EdmlDefinition definition = createDefinition(null);
+        when(schemaFetcherMock.fetchSchema(SOURCE)).thenReturn(Optional.empty());
+        final Exception exception = assertThrows(IllegalArgumentException.class,
+                () -> inferencer.inferSchema(definition));
+        assertThat(exception.getMessage(), equalTo(
+                "E-VSD-101: This virtual schema does not support auto inference for source 'mySource'. Please specify the 'mapping' element in the JSON EDML definition."));
+    }
+
+    @Test
+    void mappingNotPresentFetchingFails() {
+        final EdmlDefinition definition = createDefinition(null);
+        when(schemaFetcherMock.fetchSchema(SOURCE)).thenThrow(new RuntimeException("expected"));
+        final Exception exception = assertThrows(IllegalStateException.class, () -> inferencer.inferSchema(definition));
+        assertThat(exception.getMessage(),
+                startsWith("E-VSD-102: Schema auto inference for source 'mySource' failed. Known"));
+    }
+
+    private MappingDefinition createMapping() {
+        return Fields.builder().mapField("field", ToVarcharMapping.builder().build()).build();
+    }
+
```

Comment:
Should we add a test for successful infering the mapping?

Reply:
There should already be a test for this: `mappingNotPresent()`.